packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts

```diff
@@ -642,6 +642,10 @@ export abstract class UtxoBaseAdapter<T extends UtxoChainId> implements IChainAd
   }
 
   async parseTx(tx: unchained.utxo.types.Tx, pubkey: string): Promise<Transaction> {
+    if (!this.accountAddresses[pubkey]) {
+      await this.getAccount(pubkey)
+    }
```

Comment:
preferably-blocking: this hides the logic away and makes it look unused, prefer the below

```typescript
      const data = await this.providers.http.getAccount({ pubkey })
      
      // cache addresses for getTxHistory to use without needing to make extra requests
      this.accountAddresses[data.pubkey] = data.addresses?.map(address => address.pubkey) ?? [
        data.pubkey,
      ]
```